ui/components/multichain/connect-accounts-modal/connect-accounts-modal.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getUnconnectedAccounts } from '../../../selectors/selectors';
+import { ConnectAccountsList } from './connect-accounts-modal-list';
+
+export const ConnectAccountsModal = ({ onClose }: { onClose: () => void }) => {
+  const t = useI18nContext();
+  const accounts = useSelector(getUnconnectedAccounts);
+  const [selectedAccounts, setSelectedAccounts] = useState<string[]>([]);
+
+  const handleAccountClick = (address: string) => {
+    const index = selectedAccounts.indexOf(address);
+    let newSelectedAccounts: string[] = [];
+
+    if (index === -1) {
+      // If address is not already selected, add it to the selectedAccounts array
+      newSelectedAccounts = [...selectedAccounts, address];
+    } else {
+      // If address is already selected, remove it from the selectedAccounts array
+      newSelectedAccounts = selectedAccounts.filter(
+        (item, idx) => idx !== index,
+      );
+    }
+
+    setSelectedAccounts(newSelectedAccounts);
+  };
+
+  const selectAll = () => {
+    const newSelectedAccounts = accounts.map(
+      (account: { address: string }) => account.address,
+    );
+    setSelectedAccounts(newSelectedAccounts);
+  };
+
+  const deselectAll = () => {
+    setSelectedAccounts([]);
+  };
+
+  const allAreSelected = () => {
+    return accounts.length === selectedAccounts.length;
+  };
+  let checked = false;
+  let isIndeterminate = false;
+  if (allAreSelected()) {
+    checked = true;
+    isIndeterminate = false;
+  } else if (selectedAccounts.length > 0 && !allAreSelected()) {
+    checked = false;
```

Comment:
This isn't needed since `checked` is defaulted to `false`.

Reply:
Updated